src/main/java/io/gsonfire/postprocessors/ExclusionByValuePostProcessor.java

```diff
@@ -0,0 +1,96 @@
+package io.gsonfire.postprocessors;
+
+import com.google.gson.*;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.internal.bind.ReflectiveTypeAdapterFactory;
+import io.gsonfire.PostProcessor;
+import io.gsonfire.annotations.ExcludeByValue;
+import io.gsonfire.gson.ExclusionByValueStrategy;
+import io.gsonfire.util.FieldInspector;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+public final class ExclusionByValuePostProcessor implements PostProcessor {
+
+    private final FieldInspector fieldInspector;
+    private FieldNamingStrategy fieldNamingStrategy;
+
+    public ExclusionByValuePostProcessor(FieldInspector fieldInspector) {
+        this.fieldInspector = fieldInspector;
+        this.fieldNamingStrategy = null;
+    }
+
+    @Override
+    public void postDeserialize(Object result, JsonElement src, Gson gson) {
+        //nothing
+    }
+
+    @Override
+    public void postSerialize(JsonElement result, Object src, Gson gson) {
+        if(src == null || result.isJsonNull() || !result.isJsonObject()){
+            return;
+        }
+        for(Field f: fieldInspector.getAnnotatedFields(src.getClass(), ExcludeByValue.class)){
+            try {
+                ExcludeByValue excludeByValue = f.getAnnotation(ExcludeByValue.class);
+                Class<? extends ExclusionByValueStrategy> exclusionByValueStrategyClass = excludeByValue.value();
+
+                ExclusionByValueStrategy strategy = exclusionByValueStrategyClass.newInstance();
+                if (strategy.shouldSkipField(f.get(src))) {
+                    JsonObject resultJsonObject = result.getAsJsonObject();
+                    String fieldName = getFieldName(f, gson);
+                    if (fieldName != null) {
+                        resultJsonObject.remove(fieldName);
+                    }
+                } else {
+                    // continue
+                }
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException(e);
+            } catch (InstantiationException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private String getFieldName(Field f, Gson gson) {
+        SerializedName serializedName = f.getAnnotation(SerializedName.class);
```

Comment:
I know it might seem overkill, but reflection methods can be slow sometimes. I would see if the ```ReflectiveTypeAdapter``` class from Gson abuses the ```f.getAnnotation(SerializedName.class)``` or if it is catched.

A quick optimization would be to have a Map to cache the field serialized names. That would not take too much ram since the ```Field``` instance is always in Ram, and the total size used by all the ```String```s will be always smaller than the total size of all the ```Field```s instance.

My approach would be to build a quick Map cache, and profile the code with or without the cache.